lib/resty/lock.lua

```diff
@@ -154,7 +156,7 @@ function _M.lock(self, key)
         elapsed = elapsed + step
         timeout = timeout - step
 
-        local ok, err = dict:add(key, true, exptime)
+        local ok, err = self.dict_add(dict, key, true, exptime)
```

Comment:
Better cache `self.dict_add` in a local variable since it may get called for multiple times later on in this method.